solutions/silver/usaco-416.mdx

```diff
@@ -2,15 +2,198 @@
 id: usaco-416
 source: USACO Silver 2014 March
 title: The Lazy Cow
-author: Brad Ma, Kevin Sheng
+author: Daniel Ge, Brad Ma, Kevin Sheng
 ---
 
 [Official Analysis (C++)](http://www.usaco.org/current/data/sol_lazy_silver.html)
 
+<Spoiler title="Hint 1">
+If the range of values Bessie can move to forms a square instead of this diamond, we can apply normal prefix sums.
+</Spoiler>
+<Spoiler title="Hint 2">
+What transformation would take a diamond to a square?
+</Spoiler>
+<Spoiler title="Hint 3">
+Apply a rotation of 45 degrees on the grid such that the cells Bessie can walk to form a square.
+Note that there's one small issue that I leave to you to find and resolve (it will be covered in the solution).
+</Spoiler>
+
+<Spoiler title="Solution">
+## Explanation
+
+Apply a rotation of 45 degrees on the grid such that the cells Bessie can walk to form a square like follows:
+<table>
+<tr>
+    <td >0</td>
+    <td >0</td>
+    <td >0</td>
+    <td >0</td>
+    <td >50</td>
+    <td >0</td>
+    <td >0</td>
+    <td >0</td>
+    <td >0</td>
+</tr>
+  <tr>
+    <td >0</td>
+    <td >0</td>
+    <td >0</td>
+    <td >14</td>
+    <td >0</td>
+    <td >5</td>
+    <td >0</td>
+    <td >0</td>
+    <td >0</td>
+  </tr>
+  <tr>
+    <td >0</td>
+    <td >0</td>
+    <td >99</td>
+    <td >0</td>
+    <td >3</td>
+    <td >0</td>
+    <td >25</td>
+    <td >0</td>
+    <td >0</td>
+  </tr>
+  <tr>
+    <td >0</td>
+    <td >8</td>
+    <td >0</td>
+    <td >10</td>
+    <td >0</td>
+    <td >2</td>
+    <td >0</td>
+    <td >6</td>
+    <td >0</td>
+  </tr>
+  <tr>
+    <td >10</td>
+    <td >0</td>
+    <td >7</td>
+    <td >0</td>
+    <td >1 (B)</td>
+    <td >0</td>
+    <td >7</td>
+    <td >0</td>
+    <td >17</td>
+  </tr>
+  <tr>
+    <td >0</td>
+    <td >0</td>
+    <td >0</td>
+    <td >5</td>
+    <td >0</td>
+    <td >2</td>
+    <td >0</td>
+    <td >21</td>
+    <td >0</td>
+  </tr>
+  <tr>
+    <td >0</td>
+    <td >0</td>
+    <td >78</td>
+    <td >0</td>
+    <td >23</td>
+    <td >0</td>
+    <td >80</td>
+    <td >0</td>
+    <td >0</td>
+  </tr>
+  <tr>
+    <td >0</td>
+    <td >0</td>
+    <td >0</td>
+    <td >1</td>
+    <td >0</td>
+    <td >11</td>
+    <td >0</td>
+    <td >0</td>
+    <td >0</td>
+  </tr>
+  <tr>
+    <td >0</td>
+    <td >0</td>
+    <td >0</td>
+    <td >0</td>
+    <td >9</td>
+    <td >0</td>
+    <td >0</td>
+    <td >0</td>
+    <td >0</td>
+  </tr>
+</table>
+<br></br>
+
+What's with the zeroes?
+If we simply rotate 45 degrees and cram all the cells connected diagonally into one row, then their distance becomes $1$, leading to an awkward situation like this one.
+```
+[99, 3, 25]
+[  10, 2  ]
+[ 7, 1, 7 ]
+[   5, 2  ]
+[78, 23,80]
+```
```

Comment:
wb this one?

Reply:
I didn't really know what to do with it since it's not really an array, though I figured out a possible way now